DCOS/autoscale-testing/AutoScale.Tests.ps1

```diff
@@ -0,0 +1,271 @@
+    }
+
+    It "Has the expected number of instances in DCOS" {
+        $RG_NAME | Should not be $null
+        getScalesetsVMcount($RG_NAME) | Should be $(getDCOSagentCount($RG_NAME))
+    }
+}
+
+Describe "ScaleUp" {
+    
+    $TestCases = @()
+    $scaleSets = Get-AzureRmVmss -ResourceGroupName $RG_NAME
+    $scaleSets | Foreach-Object {$TestCases += @{scaleset = $_}}
+
+    It "Can increase the scaleset capacity" -TestCases $TestCases {
+        param($scaleset)
+        Write-Host $scaleset.Name
+        $initial_capacity = $scaleset.Sku.Capacity
+        
+        # Make sure we are initially scaled down (1 or 2 vms)
+        $initial_capacity | Should BeLessThan 3
+
+        # Scale up
+        $scaleset.Sku.capacity = 4
+        $res = Update-AzureRmVmss -ResourceGroupName $RG_NAME -Name $scaleset.Name -VirtualMachineScaleSet $scaleset
+        $res | Should not be $null
+        $res.Sku.Capacity | Should be 4
+
+        # Sanity check
+        $updated_vmss = Get-AzureRmVmss -ResourceGroupName $RG_NAME -VMScaleSetName $scaleset.Name
+        $updated_vmss.Sku.Capacity | Should be 4
+    }
+}
+
+Describe "DCOS UI" {
+    It "Reports the same amount of agents as the number of VMs in the scalesets" {
+        $RG_NAME | Should not be $null
+        $vmCount = getScalesetsVMcount($RG_NAME)
+        $agentCount = 0
+
+        $retryCount = 15
+
+        while ($retryCount -gt 0) {
+
+            $agentCount = getDCOSagentCount($RG_NAME)
+            Write-Host "Retry count=$retryCount  VMs=$vmCount  agents=$agentCount"
+            if ($vmCount -eq $agentCount){
+                break
+            }
+
+            # Only sleep if we are going to  retry
+            if ($retryCount -gt 1) {
+                Start-Sleep -Seconds 60
+            }
+
+            $retryCount -= 1
+        }
+
+        $agentCount | Should Be $vmCount
+    }
+}
+
+Describe "DCOS cli cluster" {
+
+    # $masterFQDN = getMasterFQDN($RG_NAME)
+    It "Can list the clusters" {
+        $clusters = $(dcos cluster list --json | ConvertFrom-Json)
+        $? | Should be $True
+
+        $masterFQDN = getMasterFQDN($RG_NAME)
+        $thisCluster = $clusters | Where-Object {$_.url -eq "http://$masterFQDN"}
+
+        # Only setup cluster if it not already setup or cli returns an error
+        if ($thisCluster -eq $null){
+            Write-Host "Adding DCOS cluster: $masterFQDN"
+            dcos cluster setup "http://$masterFQDN"
+            $? | Should be $True
+        }
+    }
+}
+
+Describe "DCOS service" {
+    It "Has service definition file" {
+        $path = "$here/python-server.json"
+        Test-Path $path -PathType Leaf | Should be $True
+    }
+
+
+    It "Can schedule a service" {
+        $path = "$here/python-server.json"
+        dcos marathon app add "$path"
+        $? | Should be $true
+    }
+
+}
+
+Describe "DCOS service progress" {
+
+    It "Can get the deployment" {
+        dcos marathon deployment list --json
```

Comment:
Please check if the command was successful. You only need to add the simple assertion: `$? | Should be $true`